File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.java
class Solution {
    public void solve(int[] nums, int i, List<Integer> currSubset, List<List<Integer>> subsets) {
        if (i == nums.length) {
            subsets.add(new ArrayList<>(currSubset));
            return;
        }
        currSubset.add(nums[i]);
        solve(nums, i+1, currSubset, subsets);
        currSubset.remove(currSubset.size() - 1);
        solve(nums, i+1, currSubset, subsets);
    }

    public int subsetXORSum(int[] nums) {
        List<List<Integer>> subsets = new ArrayList<>();
        List<Integer> currSubset = new ArrayList<>();
        solve(nums, 0, currSubset, subsets);
        int result = 0;
        for (List<Integer> subset : subsets) {
            int xorSum = 0;
            for (int num : subset) {
                xorSum ^= num;
            }
            result += xorSum;
        }
        return result;
    }
}

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def solve(self, nums, i, curr_subset, subsets):
        if i == len(nums):
            subsets.append(curr_subset.copy())
            return
        curr_subset.append(nums[i])
        self.solve(nums, i+1, curr_subset, subsets)
        curr_subset.pop()
        self.solve(nums, i+1, curr_subset, subsets)
        
    def subsetXORSum(self, nums: List[int]) -> int:
        subsets = []
        curr_subset = []
        self.solve(nums, 0, curr_subset, subsets)
        result = 0
        for subset in subsets:
            xor_sum = 0
            for num in subset:
                xor_sum ^= num
            result += xor_sum
        return result